src/animals/animals.service.ts

```diff
@@ -1,15 +1,175 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
-import { Model } from 'mongoose';
+import { Model, Types } from 'mongoose';
+import { RedisCacheService } from 'src/redis-cache/redis-cache.service';
+import { AnimalTypeParam } from 'src/validations/type.validator';
+import {
+  AnimalDto,
+  UpdateAnimalDto,
+  AnimalNameDto,
+  AnimalDtoResponse,
+} from './dto/animal.dto';
 import { Animal, AnimalDocument } from './schemas/animal.schema';
 
 @Injectable()
 export class AnimalsService {
   constructor(
     @InjectModel(Animal.name) private animalModel: Model<AnimalDocument>,
+    private readonly redisCacheManager: RedisCacheService,
   ) {}
 
-  async findAll() {
-    return this.animalModel.find();
+  async findAll(): Promise<AnimalDtoResponse[]> {
+    try {
+      const animals: Animal[] = await this.animalModel.find();
+      return AnimalDtoResponse.mapperArrayDto(animals);
+    } catch (err) {
+      throw new InternalServerErrorException();
+    }
+  }
+
+  async findOne(id: string): Promise<AnimalDtoResponse> {
+    try {
+      const animal: AnimalDocument = await this.animalModel.findById({
+        _id: new Types.ObjectId(id),
+      });
+
+      if (!animal) {
+        throw new NotFoundException('Animal was not found in database');
+      }
+
+      return AnimalDtoResponse.mapperDto(animal);
+    } catch (err) {
+      if (err instanceof NotFoundException) {
```

Comment:
takie cos ja bym widzial w filtrach ;)

Reply:
Ogarnąłem :)
Zmieniłem trochę podejście. Zostawiłem obsługę błędów NestJs i filtrowi, który stworzyłem. 
Dodałem odpowiednie zmiany w testach, aby sprawdzić, czy faktycznie to działa i działa :)